fiftyone/brain/internal/core/pgvector.py

```diff
@@ -0,0 +1,293 @@
+"""
+PGVector similarity backend.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import logging
+import numpy as np
+import psycopg2
+from psycopg2.errors import UndefinedColumn
+import eta.core.utils as etau
+import fiftyone.core.utils as fou
+from fiftyone.brain.similarity import (
+    SimilarityConfig,
+    Similarity,
+    SimilarityIndex,
+)
+import fiftyone.brain.internal.core.utils as fbu
+
+pgvector = fou.lazy_import("pgvector")
+logger = logging.getLogger(__name__)
+
+_SUPPORTED_METRICS = ("cosine", "dotproduct", "euclidean")
+
+
+class PgVectorSimilarityConfig(SimilarityConfig):
+    def __init__(
+        self,
+        connection_string=None,
+        metric="cosine",
+        ssl_cert=None,
+        ssl_key=None,
+        ssl_root_cert=None,
+        work_mem="64MB",  # Default work_mem value
+        **kwargs,
+    ):
+        if metric not in _SUPPORTED_METRICS:
+            raise ValueError(
+                f"Unsupported metric '{metric}'. Supported values are {_SUPPORTED_METRICS}"
+            )
+        super().__init__(**kwargs)
+        self._connection_string = connection_string
+        self.metric = metric
+        self.ssl_cert = ssl_cert
+        self.ssl_key = ssl_key
+        self.ssl_root_cert = ssl_root_cert
+        self.work_mem = work_mem
```

Comment:
We are going to need to support the user specifying the index name since they may have already created the index outside of Fiftyone. A table could actually have more than 1 embedding column and therefore would require more than 1 index. They could also create 2 indices on the same embedding column (trying different HNSW parameters) so they need to specify which one to use here. 

We need to be able to use that pre-made index but also delete that index from a fiftyone call. Since we need to do that it would be good to allow the user to specify the name of the index they want created.